client/src/components/selects/stringSelect.tsx

```diff
@@ -33,25 +33,19 @@ export default class StringSelect extends React.Component<ISelectProps, any> {
   };
 
   private filterList: ItemListPredicate<string> = (query, items) => {
-    return items
-      .filter(item => `${item.toLowerCase()}`.indexOf(query.toLowerCase()) >= 0)
-      .slice(0, this.props.maxItems || 100);
+    const stringSet = new Set(
+      items.filter(item => item.toLowerCase().startsWith(query.toLowerCase()))
+    );
```

Comment:
The objective is to add the items starting with the query first, right?
Does it solve our problem if a lot of items start with the string we want to find in the list?

Reply:
I think that the improvement to the `sortItem` function below (using string.localeCompare) solves this problem: the closest match should appear first.
Also, since the array is not sliced anymore, all matches will appear in the list (even if it's very long). To me, this is the behaviour we expect.